src/compiler/utilities.ts

```diff
@@ -3549,6 +3549,11 @@ namespace ts {
         }
         return previous[previous.length - 1];
     }
+
+    /** See comment on `declareModuleMember` in `binder.ts`. */
+    export function getCombinedLocalAndExportSymbolFlags(symbol: Symbol): SymbolFlags {
+        return symbol.exportSymbol ? symbol.exportSymbol.flags | symbol.flags : symbol.flags;

```

Comment:
In `findAllRefsForDefaultExport08` (and `findAllRefsForDefaultExport02`), we have an exported class and a non-exported namespace merged with it. So the local symbol will have flags that the exported symbol is lacking; it has `NamespaceModule|ExportValue|ExportType` while the exported symbol just has `Class`.